cf_xarray/accessor.py

```diff
@@ -1482,6 +1482,91 @@ def stack(self, dimensions=None, **dimensions_kwargs):
             dimensions.update({key: tuple(itertools.chain(*updates))})
         return self._obj.stack(dimensions)
 
+    def differentiate(
+        self,
+        coord,
+        edge_order: int = 1,
+        datetime_unit: str = None,
+        follow_positive: bool = False,
+    ):
+        """ Differentiate the array with the second order accurate central
+        differences.
+
+        .. note::
+            This feature is limited to simple cartesian geometry, i.e. coord
+            must be one dimensional.
+
+        Parameters
+        ----------
+        coord : hashable
+            The coordinate to be used to compute the gradient.
+        edge_order : {1, 2}, default: 1
+            N-th order accurate differences at the boundaries.
+        datetime_unit : {"Y", "M", "W", "D", "h", "m", "s", "ms", \
+                         "us", "ns", "ps", "fs", "as"} or None, optional
+            Unit to compute gradient. Only valid for datetime coordinate.
+        follow_positive: optional, bool
+            Fix sign of the derivative based on the ``"positive"`` attribute of ``coord``.
+            If ``positive=="down"``, then multiplied by -1.
+
+        Returns
+        -------
+        differentiated: DataArray
```

Comment:
What if we differentiate a Dataset?

Is it feasible to use a decorator to append `self._obj.differentiate` docstring?
Maybe if these kind of methods look like this it's easier to maintain them:
```python
def differentiate(self, *xr_args, follow_positive=False, **xr_kwargs):
    """
    cf_xarray docstring
                +
    appended xarray docstring
    """
```

Reply:
> append self._obj.differentiate docstring?

Could not figure this out. So i just simplified the docstring and referred the user to the avtual xarray function.s

Reply:
Sounds good!